lib/utils/object.ts

```diff
@@ -0,0 +1,215 @@
+import cloneDeepWith from 'lodash/cloneDeepWith';
+import isEqual from 'lodash/eq';
+import forOwn from 'lodash/forOwn';
+import getValue from 'lodash/get';
+import isFunction from 'lodash/isFunction';
+import isPlainObject from 'lodash/isPlainObject';
+import mergeWith from 'lodash/mergeWith';
+import { getComplexKeys } from './format';
+// eslint-disable-next-line import/order -- caused by temporarily mixing require with import
+import { camelize } from './string';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports -- .js files must be imported using require
+const baseIsNative = require('lodash/_baseIsNative');
+
+/**
+ * Deeply merges object `b` into `a`.
+ * Mutates `a`.
+ *
+ * Same concept as _.merge, but doesn't overwrite properties that have already been assigned.
+ *
+ * @param a
+ * @param b
+ */
+export function mergeDefaults<T>(a: T, b: Partial<T>): T {
+  return mergeWith(a, b, (objectValue, sourceValue) => {
+    // If it's an object, let _ handle it this time, we will be called again for each property
+    if (!isPlainObject(objectValue) && objectValue !== undefined) {
+      // _.isNative includes a check for core-js and throws an error if present.
+      // Depending on _baseIsNative bypasses the core-js check.
+      if (isFunction(objectValue) && baseIsNative(objectValue)) {
+        return sourceValue || objectValue;
+      }
+
+      return objectValue;
+    }
+
+    // eslint-disable-next-line consistent-return,no-useless-return -- lodash actually wants us to return `undefined` to fallback to the default customizer.
+    return;
+  });
+}
+
+/**
+ * An alternative to _.merge, which doesn't clone its arguments.
+ *
+ * Does not mutate parameters.
+ *
+ * Cloning is a bad idea because options arguments may contain references to sequelize
+ * models - which again reference database libs which don't like to be cloned (in particular pg-native)
+ *
+ * @param args
+ */
+export function merge(...args: object[]): object {
+  const result: { [key: string]: any } = Object.create(null);
+
+  for (const obj of args) {
+    forOwn(obj, (value, key) => {
+      if (value === undefined) {
+        return;
+      }
+
+      if (!result[key]) {
+        result[key] = value;
+      } else if (isPlainObject(value) && isPlainObject(result[key])) {
+        result[key] = merge(result[key], value);
+      } else if (Array.isArray(value) && Array.isArray(result[key])) {
+        result[key] = [...value, ...result[key]];
+      } else {
+        result[key] = value;
+      }
+    });
+  }
+
+  return result;
+}
+
+/* eslint-disable consistent-return -- lodash actually wants us to return `undefined` to fallback to the default customizer. */
```

Comment:
I think I prefer the usage of eslint-disable-next-line over eslint-disable. Makes us more aware of what we are disabling (and also makes us not forget to enable it again later)

Reply:
And I really like the explanation afterwards of why it is disabled. We should incorporate this throughout the codebase as a best practice in my opinion.

(And if we decide to do that as a best practice we should also write that down somewhere, but that's for a different PR)

Reply:
Regarding eslint-disable vs eslint-disable-next-line: Me too, but `consistent-return` is a rule that often changes where the error is reported when refactoring. It's easier to disable the whole function

If we fork eslint-plugin-eslint-comments, we could enforce these directives to go in pairs (https://mysticatea.github.io/eslint-plugin-eslint-comments/rules/disable-enable-pair.html)

Reply:
Oh and we can enforce descriptions too https://mysticatea.github.io/eslint-plugin-eslint-comments/rules/require-description.html

The fact that this page talks about v7 and later makes me think the plugin might still work with eslint 8, I'll give it a try.

Reply:
Given that these rules have been enabled, I think I can resolve this conversation now :)